ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/Transaction.java

```diff
@@ -327,6 +327,18 @@ public Wei getGasPrice() {
     return Optional.ofNullable(maxFeePerGas);
   }
 
+  public long getEffectivePriorityFeePerGas(final Optional<Long> maybeBaseFee) {
+    return maybeBaseFee
+        .filter(__ -> getType().supports1559FeeMarket())
+        .map(
+            baseFee ->
+                Math.max(
+                    0,
+                    Math.min(
+                        getMaxPriorityFeePerGas().get().getAsBigInteger().longValue(),
+                        getMaxFeePerGas().get().getAsBigInteger().longValue() - baseFee)))
```

Comment:
this is going to throw for frontier transactions post-london

Reply:
Does the `filter` above not sift those out?

Reply:
ah yes it does, quite compact 👍